File: eAutokuca/eAutokuca.Services/Database/Oprema.cs
using System;
using System.Collections.Generic;

namespace eAutokuca.Services.Database;

public partial class Oprema
{
    public int OpremaId { get; set; }

    public int? AutomobilId { get; set; }

    public bool? ZracniJastuci { get; set; }

    public bool? Bluetooth { get; set; }

    public bool? Xenon { get; set; }

    public bool? Alarm { get; set; }

    public bool? DaljinskoKljucanje { get; set; }

    public bool? Navigacija { get; set; }

    public bool? ServoVolan { get; set; }

    public bool? AutoPilot { get; set; }

    public bool? Tempomat { get; set; }

    public bool? ParkingSenzori { get; set; }

    public bool? GrijanjeSjedista { get; set; }

    public bool? GrijanjeVolana { get; set; }

    public virtual Automobil? Automobil { get; set; }
}

File: eAutokuca/UI/eautokuca_desktop/lib/models/oprema.dart
import 'package:json_annotation/json_annotation.dart';

part 'oprema.g.dart';

@JsonSerializable()
class Oprema {
  int? opremaId;
  bool? zracniJastuci;
  bool? bluetooth;
  bool? xenon;
  bool? alarm;
  bool? daljinskoKljucanje;
  bool? navigacija;
  bool? servoVolan;
  bool? autoPilot;
  bool? tempomat;
  bool? parkingSenzori;
  bool? grijanjeSjedista;
  bool? grijanjeVolana;

  Oprema(
      this.opremaId,
      this.zracniJastuci,
      this.bluetooth,
      this.xenon,
      this.alarm,
      this.daljinskoKljucanje,
      this.navigacija,
      this.servoVolan,
      this.autoPilot,
      this.tempomat,
      this.parkingSenzori,
      this.grijanjeSjedista,
      this.grijanjeVolana);

  factory Oprema.fromJson(Map<String, dynamic> json) => _$OpremaFromJson(json);

  /// Connect the generated [_$PersonToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$OpremaToJson(this);
}
